src/app/clients/add/formComponents.tsx

```diff
@@ -0,0 +1,534 @@
+    padding: 2em;
+    margin: 2em;
+    width: 100%;
+    height: 80%;
+    border-radius: 10px;
+    background-color: ${(props) => props.theme.surfaceBackgroundColor};
+    color: ${(props) => props.theme.surfaceForegroundColor};
+
+    div {
+        width: 100%;
+        margin: 0.15em 0;
+    }
+`;
+
+const Text = styled.p`
+    padding-bottom: 1em;
+`;
+
+const Subheading = styled.h2`
+    padding-bottom: 1em;
+`;
+
+const Heading = styled.h1`
+    padding-bottom: 1em;
+`;
+
+const RequiredAsterisk = styled.span`
+    color: ${(props) => props.theme.errorColor};
+
+    &:before {
+        content: "*";
+    }
+`;
+
+const ErrorText = styled(Text)`
+    color: ${(props) => props.theme.errorColor};
+    margin-bottom: 3em;
+    text-align: center;
+`;
+
+const StyledButton = styled.button`
+    text-align: center;
+    width: 150px;
+    height: 40px;
+    border-radius: 10px;
+    border: solid 0 ${(props) => props.theme.foregroundColor};
+    background-color: ${(props) => props.theme.primaryBackgroundColor};
+    color: ${(props) => props.theme.primaryForegroundColor};
+
+    &:hover {
+        background-color: ${(props) => props.theme.secondaryBackgroundColor};
+        color: ${(props) => props.theme.secondaryForegroundColor};
+    }
+`;
+
+const genericFormCard = (
+    title: string,
+    required: boolean,
+    childComponent: React.ReactNode,
+    text?: string
+): React.ReactNode => {
+    return (
+        <>
+            <StyledCard>
+                <Subheading>
+                    {title} {required && <RequiredAsterisk />}
+                </Subheading>
+                {text && <Text>{text}</Text>}
+                {childComponent}
+            </StyledCard>
+        </>
+    );
+};
+
+const initialFields: Fields = {
+    fullName: "",
+    phoneNumber: "",
+    addressLine1: "",
+    addressLine2: "",
+    addressTown: "",
+    addressCounty: "",
+    addressPostcode: "",
+    adults: [
+        { personType: "adult", quantity: 0 },
+        { personType: "male", quantity: 0 },
+        { personType: "female", quantity: 0 },
+    ],
+    numberChildren: 0,
+    children: [],
+    dietaryRequirements: {},
+    feminineProducts: {},
+    babyProducts: null,
+    nappySize: "",
+    petFood: {},
+    otherItems: {},
+    deliveryInstructions: "",
+    extraInformation: "",
+};
+
+const initialFormErrors: FormErrors = {
+    fullName: Error.initial,
+    phoneNumber: Error.none,
+    addressLine1: Error.initial,
+    addressPostcode: Error.initial,
+    adults: Error.initial,
+    numberChildren: Error.initial,
+    nappySize: Error.none,
+};
+
+const AddClientForm: React.FC = () => {
+    const router = useRouter();
+    const [fields, setFields] = useState<Fields>(initialFields);
+    const [formErrors, setFormErrors] = useState<FormErrors>(initialFormErrors);
+    const [submitError, setSubmitError] = useState<Error>(Error.none);
+
+    useEffect(() => {
+        const childrenCopy: Person[] = [];
```

Comment:
can this be static (ie not a for loop)? e.g.
```ts
if (fields.numberOfChildren > fields.children.length) {
    const extraChildren = Array(fields.numberOfChildren - fields.children.length).fill().map(item => { ... });
    fieldSetter("children", [...fields.children, ...extraChildren]);
}
if (fields.numberOfChildren < fields.children.length) {
    fieldSetter("children", fields.children.slice(fields.children.length));
}
```
(note: all untested code, just what I think _could_ work!)